.github/workflows/integration.yaml

```diff
@@ -28,46 +29,31 @@ jobs:
     - name: Check out repo
       uses: actions/checkout@v2
 
-    - uses: balchua/microk8s-actions@v0.2.2
+    - name: Setup operator environment
+      uses: charmed-kubernetes/actions-operator@main
       with:
-        channel: '1.20/stable'
-        addons: '["dns", "storage", "rbac", "metallb:10.64.140.43-10.64.140.49"]'
+        provider: microk8s
+        charmcraft-channel: latest/candidate
 
     - name: Install dependencies
       run: |
         set -eux
-        sudo snap install charm --classic
-        sudo snap install juju --classic
-        sudo snap install juju-bundle --classic
         sudo snap install juju-wait --classic
-        sudo snap install charmcraft --classic --channel latest/candidate
         sudo apt update
         sudo apt install -y firefox-geckodriver tox
 
-    # Avoid race condition with storage taking a long time to initialize
-    - name: Wait for storage
-      run: |
-        sg microk8s -c 'microk8s kubectl rollout status deployment/hostpath-provisioner -n kube-system'
+    - name: Ensure storage is up
+      run: kubectl rollout status deployment/hostpath-provisioner -n kube-system
 
-    - name: Bootstrap
-      run: |
-        set -eux
-        sg microk8s -c 'juju bootstrap microk8s uk8s'
-        juju add-model mlflow
+    - run: sg microk8s -c 'juju bootstrap microk8s uk8s'
+    - run: juju add-model mlflow
 
     - name: Deploy charm dependencies
       timeout-minutes: 15
       run: |
         set -eux
-        juju model-config update-status-hook-interval=15s
-        juju deploy ch:istio --trust
-        juju config istio-pilot 'default-gateways=kubeflow-gateway'
-        sleep 10
-        juju wait -wvt 300 --retry_errors 20
-        juju model-config update-status-hook-interval=5m
-
-        # https://bugs.launchpad.net/juju/+bug/1921739
-        (i=600; while ! juju wait -wvt 30 ; do ((--i)) || exit; sleep 1; done)
+        juju deploy istio --channel 1.11/stable --trust
```

Comment:
Should we be using `1.11` for this?

Reply:
Until https://bugs.launchpad.net/juju/+bug/1957619 is fixed, we don't have much of a choice. Deploying 1.5 requires some manual tweaks.